src/plugins/multipart.js

```diff
@@ -93,6 +93,13 @@ function createInitMultipart(boundary) {
         headerField = '';
         headerValue = '';
       } else if (name === 'headersEnd') {
+        // if filename exists but mimetype is undetermined, 
+        // use file-extension to determine the mimetype
+        // works for zip, pdf, json
+        if (!part.mimetype && typeof part.originalFilename === 'string') {
+          const extension = part.originalFilename.split('.').slice(-1)[0];
+          part.mimetype = `application/${extension}`;
+        }
```

Comment:
I like where this is going, but I'd like to see a few other things:

* If there's no file extension, it seems like this might end up with `application/undefined`. It seems like `application/octet-stream` would be most appropriate for a file of unknown mime type as per RFC:

> [4.4. Content-Type Header Field for Each Part](https://www.rfc-editor.org/rfc/rfc7578#section-4.4)
> ```
>    Each part MAY have an (optional) "Content-Type" header field, which
>    defaults to "text/plain".  If the contents of a file are to be sent,
>    the file data SHOULD be labeled with an appropriate media type, if
>    known, or "application/octet-stream".
> ```

* The `application/<extension>` is alright, but a bit fragile. It seems like a map of common file extensions to mime types would accurately cover core use cases. Maybe just source from something like: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

* Log any unrecognized file extensions and fallback to `application/octet-stream`